linker/Mono.Linker/DirectoryAssemblyResolver.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Collections.Generic;
+using Mono.Cecil;
+
+#if NET_CORE
+namespace Mono.Linker {
+
+	public sealed class AssemblyResolutionException : FileNotFoundException {
+
+		readonly AssemblyNameReference reference;
+
+		public AssemblyNameReference AssemblyReference {
+			get { return reference; }
+		}
+
+		public AssemblyResolutionException (AssemblyNameReference reference)
+			: base (string.Format ("Failed to resolve assembly: '{0}'", reference))
+		{
+			this.reference = reference;
+		}
+	}
+
```

Comment:
This may conflict with my PR that was recently merged: https://github.com/mono/linker/pull/16/ . In that PR I'm catching AssemblyResolutionException, which is now ambiguous. Can't we make Mono.Cecil.AssemblyResolutionException available under NET_CORE and use it here?